### PR TITLE
Update Crossref REST translator

### PR DESCRIPTION
**Pull request by mrtcode**
I noticed there are some differences in the available data compared to the XML API. For example, fields like ISBN, pages, editors, contributors, and language, etc. can sometimes be missing, but not always. However, this translator performs better with various book-related items.

It was tested with all item types provided by the Crossref API, and each item type is also included in the tests.

All in all, I think it's pretty good now.

**The hyphen has been removed from the translator file name to match the actual translator name (there was an issue with Scaffold). I wonder if this could have any negative consequences?**

### CODE REVIEW

**Comment by mrtcode** [comment]
To run test with Scaffold `detectSearch` has to be changed to return `true`.

**Comment by dstillman** [comment]
We should set Library Catalog to just "Crossref", not "Crossref REST".

File: Crossref REST.js [hunk 1]
```diff
@@ -0,0 +1,1733 @@
...
+			item.date = parseDate(result.issued);
+		}
+
+		if (result.DOI) {
+			if (ZU.fieldIsValidForType('DOI', item.itemType)) {
+				item.DOI = result.DOI;
+			}
+			// Add DOI to extra for unsupported items
+			else {
+				item.extra = (item.extra ? item.extra + '\n' : '') + 'DOI: ' + result.DOI;
```
Comment (dstillman) [inline Crossref REST.js:307]: We shouldn't need to do this manually. It should get set automatically if not valid for the field.

Comment (dstillman) [reply Crossref REST.js:307]: (But that might not show up in tests.)

Comment (mrtcode) [reply Crossref REST.js]: Yes, it was exactly like that.

**Comment by mrtcode** [comment]
> We should set Library Catalog to just "Crossref", not "Crossref REST".

Fixed. However, I imagine it will be [overridden](https://github.com/zotero/translators/blob/c140b805886f37e6021b8b62925e41f23f994007/DOI%20Content%20Negotiation.js#L82), as is the case with the XML translator.

**Pull request merged**